code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm

```diff
@@ -152,5 +150,591 @@ datum/reagent/medicine/spaceacillin
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.25 * REAGENTS_METABOLISM
+
+datum/reagent/medicine/perfluorodecalin/on_mob_life(var/mob/living/carbon/human/M as mob)
+	M.adjustOxyLoss(-12*REM)
+	M.silent = max(M.silent, 5)
+	if(prob(33))
+		M.adjustBruteLoss(-0.5*REM)
+		M.adjustFireLoss(-0.5*REM)
+	..()
+	return
+
+datum/reagent/medicine/ephedrine
+	name = "Ephedrine"
+	id = "ephedrine"
+	description = "Reduces stun times, increases run speed. If overdosed it will deal toxin and oxyloss damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	overdose_threshold = 45
+	addiction_threshold = 30
+
+datum/reagent/medicine/ephedrine/on_mob_life(var/mob/living/M as mob)
+	M.status_flags |= IGNORESLOWDOWN
+	M.AdjustParalysis(-1)
+	M.AdjustStunned(-1)
+	M.AdjustWeakened(-1)
+	M.adjustStaminaLoss(-1*REM)
+	..()
+	return
+
+datum/reagent/medicine/ephedrine/overdose_process(var/mob/living/M as mob)
+	if(prob(33))
+		M.adjustToxLoss(0.5*REM)
+		M.losebreath++
+	..()
+	return
+
+datum/reagent/medicine/ephedrine/addiction_act_stage1(var/mob/living/M as mob)
+	if(prob(33))
+		M.adjustToxLoss(2*REM)
+		M.losebreath += 2
+	..()
+	return
+datum/reagent/medicine/ephedrine/addiction_act_stage2(var/mob/living/M as mob)
+	if(prob(33))
+		M.adjustToxLoss(3*REM)
+		M.losebreath += 3
+	..()
+	return
+datum/reagent/medicine/ephedrine/addiction_act_stage3(var/mob/living/M as mob)
+	if(prob(33))
+		M.adjustToxLoss(4*REM)
+		M.losebreath += 4
+	..()
+	return
+datum/reagent/medicine/ephedrine/addiction_act_stage4(var/mob/living/M as mob)
+	if(prob(33))
+		M.adjustToxLoss(5*REM)
+		M.losebreath += 5
+	..()
+	return
+
+datum/reagent/medicine/diphenhydramine
+	name = "Diphenhydramine"
+	id = "diphenhydramine"
+	description = "Purges body of lethal Histamine and reduces jitteriness while causing minor drowsiness."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+
+datum/reagent/medicine/diphenhydramine/on_mob_life(var/mob/living/M as mob)
+	if(prob(50))
+		M.drowsyness += 1
+	M.jitteriness -= 1
+	M.reagents.remove_reagent("histamine",1.5)
+	..()
+	return
+
+datum/reagent/medicine/morphine
+	name = "Morphine"
+	id = "morphine"
+	description = "Will allow you to ignore slowdown from equipment and damage. Will eventually knock you out if you take too much. If overdosed it will cause jitteriness, dizziness, force the victim to drop items in their hands and eventually deal toxin damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	var/cycle_count = 0
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	overdose_threshold = 30
+	addiction_threshold = 25
+
+
+datum/reagent/medicine/morphine/on_mob_life(var/mob/living/M as mob)
+	M.status_flags |= IGNORESLOWDOWN
+	if(cycle_count >= 36)
+		M.sleeping += 3
+	cycle_count++
```

Comment:
Why did you remove my cycle_count variable from the base reagents thing? Please move var/cycle_count back up to all reagents and make it increase by 1 every time on_mob_life() is called.